op-devstack/sysgo/l2_el.go

```diff
@@ -37,6 +58,60 @@ func (n *L2ELNode) hydrate(system stack.ExtensibleSystem) {
 	l2Net.(stack.ExtensibleL2Network).AddL2ELNode(sysL2EL)
 }
 
+func (n *L2ELNode) Start() {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+	if n.l2Geth != nil {
+		n.logger.Warn("op-geth already started")
+		return
+	}
+
+	require := n.p.Require()
+	l2Geth, err := geth.InitL2(n.id.String(), n.l2Net.genesis, n.jwtPath,
+		func(ethCfg *ethconfig.Config, nodeCfg *gn.Config) error {
+			ethCfg.InteropMessageRPC = n.supervisorRPC
+			ethCfg.InteropMempoolFiltering = true
+			nodeCfg.P2P = p2p.Config{
+				NoDiscovery: true,
+				ListenAddr:  "127.0.0.1:0",
+				MaxPeers:    10,
+			}
+			if n.authRPC != "" {
+				// Preserve the existing auth rpc port
+				nodeCfg.AuthPort = rpcPort(require, n.authRPC)
+			}
+			if n.userRPC != "" {
+				// Preserve the existing websocket rpc port
+				nodeCfg.WSPort = rpcPort(require, n.userRPC)
+			}
+			return nil
+		})
+	require.NoError(err)
+	require.NoError(l2Geth.Node.Start())
+	n.l2Geth = l2Geth
+	n.authRPC = l2Geth.AuthRPC().RPC()
+	n.userRPC = l2Geth.UserRPC().RPC()
+}
+
+func rpcPort(require *testreq.Assertions, rpc string) int {
+	u, err := url.Parse(rpc)
+	require.NoError(err, "Failed to parse existing rpc url")
+	port, err := strconv.Atoi(u.Port())
+	require.NoError(err, "Invalid rpc port")
+	return port
+}
+
+func (n *L2ELNode) Stop() {
+	if n.l2Geth == nil {
```

Comment:
guard this function using the `n.mu` lock. Or make it private to ensure it's only at testing cleanup.

Reply:
Yep should have been locked - it's part of an interface (called by ControlPlan) so can't be private.